src/filters/filter_conversion.hpp

```diff
@@ -403,6 +404,223 @@ COMPILER_SUPRESS("-Weffc++")
+                map.float_channels[name + ":value"] = &output;
+            }
+        }
+
+        virtual void update() override {
+            double val = (*input - lower_bound_in) / (upper_bound_in - lower_bound_in) * (upper_bound_out - lower_bound_out) + lower_bound_out;
+            if constexpr (std::is_same<T, uint8_t>::value) {
+                this->output = (uint8_t) std::round(std::max(std::min(val, 255.0), 0.0));
+            } else if constexpr (std::is_same<T, uint16_t>::value) {
+                this->output = (uint16_t) std::round(std::max(std::min(val, 65535.0), 0.0));
+            } else if constexpr (std::is_same<T, double>::value) {
+                this->output = val;
+            }
+        }
+
+        virtual void scene_activated() override {}
+
+    };
+
+    using filter_float_map_range_8bit = float_map_range<uint8_t, filter_type::filter_float_map_range_8bit>;
+    using filter_float_map_range_16bit = float_map_range<uint16_t, filter_type::filter_float_map_range_16bit>;
+    using filter_float_map_range_float = float_map_range<double, filter_type::filter_float_map_range_float>;
+
+    class filter_dual_byte_to_16bit : public filter {
+    private:
+        uint16_t output = 0;
+        uint8_t* lower_input = nullptr;
+        uint8_t* upper_input = nullptr;
+    public:
+        filter_dual_byte_to_16bit() : filter() {}
+        virtual ~filter_dual_byte_to_16bit() {}
+
+        virtual void setup_filter(const std::map<std::string, std::string>& configuration, const std::map<std::string, std::string>& initial_parameters, const channel_mapping& input_channels, const std::string& own_id) override {
+            MARK_UNUSED(initial_parameters);
+            MARK_UNUSED(configuration);
+            if(!input_channels.eight_bit_channels.contains("lower")) {
+                throw filter_config_exception("Unable to link input of 8 bit merge to 16 bit filter: channel mapping does "
+                                              "not contain channel 'lower' of type 'uint8_t'.",
+                                              filter_type::filter_dual_byte_to_16bit, own_id);
+            }
+            this->lower_input = input_channels.eight_bit_channels.at("lower");
+            if(!input_channels.eight_bit_channels.contains("upper")) {
+                throw filter_config_exception("Unable to link input of 8 bit merge to 16 bit filter: channel mapping does "
+                                              "not contain channel 'upper' of type 'uint8_t'.",
+                                              filter_type::filter_dual_byte_to_16bit, own_id);
+            }
+            this->upper_input = input_channels.eight_bit_channels.at("upper");
+        }
+
+        virtual bool receive_update_from_gui(const std::string& key, const std::string& _value) override {
+            MARK_UNUSED(key);
+            MARK_UNUSED(_value);
+            return false;
+        }
```

Comment:
As something to think about: Should we make the output range configurable? The possible use case would be switching between effect ranges. Most likely adding an adder with constant after the required offset would solve this issue though.